src/osd/PG.cc

```diff
@@ -988,8 +992,14 @@ map<pg_shard_t, pg_info_t>::const_iterator PG::find_best_info(
     if (p->second.last_update < min_last_update_acceptable)
       continue;
     // disqualify anyone with a too old last_epoch_started
-    if (p->second.last_epoch_started < max_last_epoch_started_found)
-      continue;
```

Comment:
in what situation, last_epoch_clean is bigger but last_epoch_started is smaller?   if last_epoch_clean is bigger means already finish recovery, so last_epoch_started is already updated (peering is finished)